### PR TITLE
[psud] Increase unit test coverage; Refactor mock platform

### PR DESCRIPTION
**Pull request by jleveque**
#### Description
- Refactor psud such that the `run()` method does not contain an infinite loop, thus allowing us to unit test it
- Refactor mock_platform.py such that it inherits from sonic-platform-common in order to ensure it is aligned with the current API definitions (this introduces a test-time dependency on the sonic-platform-common package)
- Eliminate the need to check for a `PSUD_UNIT_TESTING` environment variable in daemon code
- Increase overall unit test unit test coverage from 78% to 97%

Previous unit test coverage:

```
----------- coverage: platform linux, python 3.7.3-final-0 -----------
Name           Stmts   Miss  Cover
----------------------------------
scripts/psud     360     81    78%
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml
```

Unit test coverage with this patch:

```
----------- coverage: platform linux, python 3.7.3-final-0 -----------
Name           Stmts   Miss  Cover
----------------------------------
scripts/psud     366     12    97%
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml

```

#### Motivation and Context

To increase unit test coverage >= 90% in order to prevent regressions

#### How Has This Been Tested?

The refactored psud has been tested on a physical DUT and the unit test results can be examined via the Azure Pipelines check builds in this PR.

### CODE REVIEW

**Comment by jleveque** [comment]
@mprabhu-nokia: I have done a bit more refactoring to psud in order to increase unit test coverage. Can you please test and review these changes on a modular chassis?

**Comment by jleveque** [comment]
Unit test is failing due to a bug uncovered in sonic_platform_base. Fix here: https://github.com/Azure/sonic-platform-common/pull/173

File: sonic-psud/tests/test_PsuChassisInfo.py [hunk 1]
```diff
@@ -189,17 +197,17 @@ def test_consumed_power(self):
 
     def test_power_budget(self):
         chassis = MockChassis()
-        psu = MockPsu(True, True, "PSU 1", 0)
+        psu1 = MockPsu("PSU 1", 0, True, True)
         psu1_power = 510.0
         psu.set_maximum_supplied_power(psu1_power)
```
Comment (mprabhu-nokia) [inline sonic-psud/tests/test_PsuChassisInfo.py:202]: Line #194, object is psu1..so, should be psu1.set_maximum_supplied_power()? Please check other instances like line #223

Comment (jleveque) [reply sonic-psud/tests/test_PsuChassisInfo.py]: Good catch, thanks! Fixed in latest commit.

**Comment by jleveque** [comment]
@mprabhu-nokia: Can you please review and test this PR?

**Comment by jleveque** [comment]
@mprabhu-nokia: Can you please review and test this PR?

**Review by mprabhu-nokia — APPROVED** [review]
Looks good.

**Pull request merged**